sentry-kotlin-multiplatform/src/commonMain/kotlin/io/sentry/kotlin/multiplatform/SentryBaseEvent.kt

```diff
@@ -0,0 +1,52 @@
+package io.sentry.kotlin.multiplatform
+
+import io.sentry.kotlin.multiplatform.protocol.Breadcrumb
+import io.sentry.kotlin.multiplatform.protocol.SentryId
+import io.sentry.kotlin.multiplatform.protocol.User
+
+/** Base class for all Sentry events. */
+public abstract class SentryBaseEvent(public open var eventId: SentryId = SentryId.EMPTY_ID) {
+
+    public open var release: String? = null
+    public open var environment: String? = null
+    public open var platform: String? = null
+    public open var user: User? = null
+    public open var serverName: String? = null
+    public open var dist: String? = null
+
+    /** This is not thread-safe */
+    public open val contexts: Map<String, Any>? get() = mutableContexts
+    internal var mutableContexts: Map<String, Any>? = null
+
+    /** This is not thread-safe */
+    public open val breadcrumbs: List<Breadcrumb>? get() = mutableBreadcrumbs
+    internal var mutableBreadcrumbs: MutableList<Breadcrumb>? = null
```

Comment:
Why do we need this instead of doing it directly?

Reply:
You're right, we don't need this for breadcrumbs, tags, fingerprints and exceptions.

However, I'd like to keep it for context to keep context immutable as I'll introduce a separate `Context` class later on to modify context much more easily with less code duplication.

Reply:
It's not immutable, the getter just returns the `mutableBreadcrumbs`, the only difference is the interface, or am I missing something? If you want something immutable, you need an immutable type such as.
`val immutableList: List<String> = mutableList.toList()`

Reply:
Yes it's not truly immutable but it prevents the user from modifying it by exposing only a `List` so to be exact we are exposing a read-only List, not an immutable one.

However, we don't need that anymore as we can also do this:

```
public var contexts: Map<String, Any>? = mapOf()
    internal set
```

Since we only care about being able to set the context internally but not publicly.